Example/Emission/UnroutedViewController.m

```diff
@@ -1,49 +1,106 @@
 #import "UnroutedViewController.h"
 #import <Artsy+UIFonts/UIFont+ArtsyFonts.h>
 #import <FLKAutoLayout/FLKAutoLayout.h>
+#import <Emission/AREmission.h>
+#import <Emission/ARSwitchBoardModule.h>
 
-@interface UnroutedViewController()
-@property NSString *route;
+#import "ARDefaults.h"
+
+@import WebKit;
+
+@interface UnroutedViewController() <WKNavigationDelegate>
+@property (readonly, nonatomic, strong) WKWebView *webView;
+@property (nonatomic, readonly, strong) NSURL *initialURL;
 @end
 
 @implementation UnroutedViewController
 
-- (instancetype)initWithRoute:(NSString *)title
+- (instancetype)initWithRoute:(NSString *)route
 {
   if ((self = [super init])) {
-    self.route = title;
+    _initialURL = [self urlForRoute:route];
+    self.automaticallyAdjustsScrollViewInsets = NO;
+  }
+
+  return self;
+}
+
+- (NSURL *)urlForRoute:(NSString *)route
+{
+  if ([route containsString:@"https"]) {
+    return [NSURL URLWithString:route];
   }
-  return  self;
+
+  BOOL useStaging = [[NSUserDefaults standardUserDefaults] boolForKey:ARUseStagingDefault];
+  NSURL *url = [NSURL URLWithString: useStaging ? @"https://staging.artsy.net" : @"https://artsy.net"];
+  return [url URLByAppendingPathComponent:route];
+}
+
+- (void)loadURL:(NSURL *)URL;
+{
+  AREmission *emission = [AREmission sharedInstance];
+  NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:URL];
+  NSString *agentString = [NSString stringWithFormat:@"Mozilla/5.0 Artsy-Mobile/3.3.0 Eigen/2017.07.07.13/3.3.0 (iPhone; iOS 9.0; Scale/2.00) AppleWebKit/601.1.46 (KHTML, like Gecko) Emission/1.x"];
+
+  [request addValue:agentString forHTTPHeaderField:@"UserAgent"];
+  [request addValue:emission.configurationModule.userID forHTTPHeaderField:@"X-User-ID"];
+  [request addValue:emission.configurationModule.authenticationToken forHTTPHeaderField:@"X-Access-Token"];
+
+  [self.webView loadRequest:request];
 }
 
 - (void)viewDidLoad
 {
-  UILabel *label = [UILabel new];
-  label.text = self.route;
-  label.font = [UIFont serifFontWithSize:18];
-  label.textAlignment = NSTextAlignmentCenter;
-  label.numberOfLines = 0;
+  [super viewDidLoad];
+  WKWebViewConfiguration *config = [[WKWebViewConfiguration alloc] init];
+
+  CGRect deviceBounds = [UIScreen mainScreen].bounds;
```

Comment:
Maybe this version with auto-layout can work better?

``` objc
webView.translatesAutoresizingMaskIntoConstraints = NO;
  [self.view addConstraints:@[
    [NSLayoutConstraint constraintWithItem:webView
                                 attribute:NSLayoutAttributeTop
                                 relatedBy:NSLayoutRelationEqual
                                    toItem:self.topLayoutGuide
                                 attribute:NSLayoutAttributeBottom
                                multiplier:1
                                  constant:0],
    [NSLayoutConstraint constraintWithItem:webView
                                 attribute:NSLayoutAttributeLeading
                                 relatedBy:NSLayoutRelationEqual
                                    toItem:self.view
                                 attribute:NSLayoutAttributeLeading
                                multiplier:1
                                  constant:0],
    [NSLayoutConstraint constraintWithItem:webView
                                 attribute:NSLayoutAttributeTrailing
                                 relatedBy:NSLayoutRelationEqual
                                    toItem:self.view
                                 attribute:NSLayoutAttributeTrailing
                                multiplier:1
                                  constant:0],
    [NSLayoutConstraint constraintWithItem:webView
                                 attribute:NSLayoutAttributeBottom
                                 relatedBy:NSLayoutRelationEqual
                                    toItem:self.bottomLayoutGuide
                                 attribute:NSLayoutAttributeTop
                                multiplier:1
                                  constant:0]
    ]];

```

Reply:
Using frames is fine especially in this case. Way less verbose too.

Reply:
The below version is a bit less messier ( three lines that are descriptive )

```objc
  [self.webView constrainTopSpaceToView:self.flk_topLayoutGuide predicate:@"0"];
  [self.webView alignLeading:@"0" trailing:@"0" toView:self.view];
  [self.webView alignBottomEdgeWithView:self.view predicate:@"0"];
```

So I'll stick with that for now 👍

Reply:
Oh yeah, using FLK to cleanup that auto layout code is better, but frame code would be inconsistent with the rest of the app.